### DISCUSSION TITLE
Action to get file from own package when running inside a workflow

### DISCUSSION THREAD

**Discussion by plocket** [community/community#32593] [Actions] [answered]
I'd like my action to copy a file from its own directory, even if it's running in a workflow of another repository. Not sure how to describe this, so I'll try a couple different ways.

I am making an action that runs in the workflows of my users' repos. When a user triggers the action inside their own repo, the action can access the files in the user's repo just fine, but that's not what I want. Is it possible for my action to read/copy the file of _my_ package - the package that the action itself is in.

My repo dir:
```
action.yml
file-to-copy.json
```

User's repo dir:
```
.github
  workflows
    some-workflow.yml
super-cool-project.js
```

Given the above folder structures, the user's `some-workflow.yml` will use my repo's `action.yml`. Can that `action.yml` copy the contents of `file-to-copy.json`?

**Comment by airtower-luna** [comment, marked as answer]
Probably, but how to do it depends on the type of action.

* For a composite action, you can use `github.action_path` to find your repository.
* For a Docker action, make the Dockerfile copy the file into the image and use it from there during runtime.
* For a Javascript action it looks like using `github.action_repository` and `github.action_ref` to download the file appears to be the best option.

See [`github` context](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context), the docs mention `github.action_path` is only available for composite actions.

**Reply by plocket** [reply]
`github.action_path` was perfect, thank you. Ended up doing the following to copy a file from the action repo to the repo with the workflow using the action:

```yml
runs:
  using: "composite"
  steps:
    - name: make file
      run: |
        touch new_file.js
        cat "${{ github.action_path }}/my_file.js" > new_file.js
      shell: bash
```
